python_modules/dagster/dagster/_config/structured_config.py

```diff
@@ -27,7 +27,34 @@ class cached_property:  # type: ignore[no-redef]
 from dagster._core.storage.io_manager import IOManager, IOManagerDefinition
 
 
-class Config(BaseModel):
+class MakeConfigCacheable(
```

Comment:
maybe `CacheableBaseModel`? or something like `ConfigBase`

Reply:
I am against a *Base naming convention as it is extremely unclear why the extra class exists.  `MakeConfigCacheable` is at a minimum very clear. I can change `CacheableBaseModel` but I actually don't like too much either as no one else is supposed to inherit form this. `MakeConfigCacheable` indicates the sole purpose of the class is to make the `Config` cache-able and that is all.

Reply:
keeping it as `MakeConfigCacheable` seems fine to me